MdeModulePkg/Application/UiApp/FrontPage.c

```diff
@@ -667,6 +674,19 @@ UpdateFrontPageBannerStrings (
     Status = Smbios->GetNext (Smbios, &SmbiosHandle, NULL, &Record, NULL);
   }
 
+  if ( InstalledMemory == 0 && MemorySize != 0xFFFF ) {

```

Comment:
The problem with these changes is that type17 and type19 might not always be in ascending order (hypothetically). If both types are present, we will have double the real memory capacity in the system (a sum of type17 and type19) if it is in ascending order (type17 will be counted first, then type 19 will add up more memory unconditionally). Type17 should be used as a fallback and parsed separately in a different loop or a different condition should be used to prevent such scenario.

Reply:
Right, sorry @SergiiDmytruk 

It is hard to notice the scopes in the GH web GUI diff in the PR... Closing the thread.